cmd/doctor.go

```diff
@@ -124,13 +125,18 @@ func runRecreateTable(ctx *cli.Context) error {
 }
 
 func runDoctor(ctx *cli.Context) error {
+	stdCtx, cancel := installSignals()
+	defer cancel()
+
+	// some doctor sub-commands need to use git command
+	if err := git.InitWithSync(stdCtx); err != nil {
+		return err
+	}
```

Comment:
Shouldn't this happen only after the logging is set up? eg. after line 163

Reply:
I do not want to put the code after `log.DelNamedLogger`. Otherwise there will be more bugs like: 

* https://github.com/go-gitea/gitea/issues/12082